tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/TypeSuite.scala

```diff
@@ -98,14 +98,47 @@ class TypeSuite extends BaseDottySuite {
   }
 
   test("coloneol-type3") {
-    runTestError[Stat](
+    runTestAssert[Stat](
       """|type A = Product:
```

Comment:
This doesn't seem to compile with Scala 3:
```
type A = Product:
[error]    |                  ^
[error]    |                  end of statement expected but ':' found
```

I don't think we can have colon for refined types?

Reply:
interesting... i found the Refinement definition here https://dotty.epfl.ch/docs/internals/syntax.html, not sure if I interpreted it correctly. or is it not out yet?

Reply:
in fact, there's no mention of `with` to refine...

Reply:
Not sure, I will ask around.

Reply:
It seems this will work from 3.3.x so this might have been a recent change.